src/_statusline-semaphore.ts

```
@@ -0,0 +1,180 @@
+function ensureSemaphoreDir(): void {
+	const dir = getSemaphoreDir();
+	if (!existsSync(dir)) {
+		mkdirSync(dir, { recursive: true });
+	}
+}
+
+/**
+ * Gets the semaphore file path for a session
+ */
+function getSemaphoreFilePath(sessionId: string): string {
+	return join(getSemaphoreDir(), `${sessionId}.json`);
```

ESLint

[error] Unsafe return of a value of type error.

(ts/no-unsafe-return)